### DISCUSSION TITLE
AI/ML course discussion

### DISCUSSION THREAD

**Discussion by pk1098** [snigi-gupta/AttendingUniversityAtBuffalo#2] [Q&A] [answered]
Hey, my name is Prashant Kalyani. I attended the webinar where you discussed different courses and I have few questions. I am planning to take Intro to ML this spring sem online so can you guide me that which concepts should I revise or what are prerequisites that I should cover before the classes start

**Comment by snigi-gupta** [comment, marked as answer]
Hi @pk1098,

Thank you for your question. While I took IML by Sargur Srihari, I think the material of the course will not differ by much. IML is highly python heavy. So I will definitely suggest you brush up on basic Python Syntax mainly - [List comprehension, Dict comprehension, Slicing, Conditional statements]. Python is very language friendly, so I think picking up on the language will be fairly easy. 

Apart from this, you will need to brush up on some python libraries. Namely - [Pandas, Keras, PyTorch, numPy, matplotlib, sklearn] These libraries are python based but they do have a learning curve. Always read the documentation for a better understanding. Their communities are also very helpful. I also referred widely to blogs on Medium, TowardsAI, and TowardsDataScience. These ML/AI blogs guide you through every step and deeply help you in your understanding. 

Pro Tip: Open these blogs in incognito mode for the full article.

As for the course structure, you will be taught Logistic Regressions, Bath Normalization, Supervised and Unsupervised learning and also touch through Reinforcement Learning. 

I hope this helps. If you have follow-up questions, don't hesitate to ask!

**Reply by pk1098** [reply]
It was very helpful, thank you. I am planning to pursue AI/ML specialization so what are the different courses that you would recommend, considering the job prospects in the USA.

**Reply by snigi-gupta** [reply]
@pk1098 ,

If you have no experience and are planning to take a job in ML - it is very unlikely that you will be considered for the position. In fact, if you have SDE experience of some years, even then ML/AI jobs in the USA do not consider MS students. Even if the job description says MS/PhD students desired, a call back is hardly received by an MS student.

I would suggest you rethink your goals first. If the purpose of your MS is to get a job in the USA, then you should look for an SDE specialization. Additionally, you should back-track your goals. Look at it this way, after 1.5 years of MS, where do you want to work, what do you want to work in? Even if you do want to go into an ML-based role, then you can join as an SDE in a company and later on internally pave your path towards ML/AI-based roles. That is a much better route I believe.

Having said that, if you do have some ML experience in your undergraduate or have working on ML technologies in some company, you can choose Reinforcement Learning, Computer Vision, Natural Language Processing, Robotics, Independent Study with leading ML/AI professors at UB. You can check about professors' work and the subjects they teach by going to their UB page of the website. I would also highly encourage to build self-projects on Kaggle or using ML/AI python libraries such as OpenCV, Pytorch etc.

Lastly, I would encourage you to pursue a Ph.D. if you are really passionate about ML/AI. This is because once you pursue a PhD. a lot more companies and opportunities will open up to you. Most ML/AI based companies hire Ph.D. students. Even for internships. But do remember Ph.D is a huge commitment so be very clear before signing up for it!

**Reply by shalini-agarwal** [reply]
@snigi-gupta Hello, this might be a naive question but since I am from a non-cs background (though I have a work ex in this field), I wanted to understand what topics/courses should a person take if she is intending to work as SDE? And how different is (courses/topics) it as compared to the role of ML engineer?
2. I do have a work ex of 1.9 years in AI (NLP, CV, and a little ML) and I have planned on working in this domain itself after completing my MS but after reading your comments above about how difficult it is to get a job in this field, I am re-considering. So if I take up courses in my MS which builds a specialisation in AI, but in the end apply for SDE roles, will I be at some disadvantage as compared to students who, from the start, took courses inclined towards SDE roles?

**Reply by snigi-gupta** [reply]
@shalini-agarwal replied to your question in the [YTD github discussion section](https://github.com/snigi-gupta/yourtechdiary/discussions/7)

**Comment by poojitha66** [comment]
Hi @snigi-gupta ,

I hope this message finds you well. I recently attended the university webinar where coordinators discussed the two tracks available for our course: course track and research track. I have already selected the course track and within that, we have the option to choose between AI/ML and systems tracks.

Considering the current job requirements and trends in the industry, I kindly request your guidance on which track would be better to pursue. Below are the track courses provided by the university for Fall -2023

**Fall Entry, 1st semester (12 required credits)**

CSE 521: Operating Systems
CSE 531: Algorithms
CSE 574: Introduction to ML
CSE 587: Data Intensive Computing

**Fall Entry, 2nd semester (9 required credits)**

one Elective
CSE 565: Computer Security
AI/ML: CSE 676: Deep Learning; CSE 560: Data Models & Query Languages
Systems: CSE 589: Network Concepts; CSE 590: Computer Architecture

****Fall Entry, 3rd semester (6 required credits)**

one Elective
AI/ML: CSE 573: Computer Vision
Systems: CSE 562: Database Systems

All students take the 27 required credits (same non-elective courses)


I understand that both tracks have their merits, but I believe your expertise and insights will help me make an informed decision. Your valuable input regarding the suitability of each track in relation to the current job market and industry demands would be greatly appreciated. I am eager to enhance my skills in a direction that aligns with the evolving needs of the industry.

Thank you in advance for your time and consideration. I look forward to hearing your expert advice and making an informed decision that will contribute to my professional growth.

Best regards,
Poojitha

**Comment by snigi-gupta** [comment]
Hi @poojitha66 ,

I understand that you expect me to tell you the answer to your question, choosing a track is a very personal decision. 

As you mentioned, you should look at what your interests are, what the future of tech looks like, what you think you would love to learn and do. You can excel in either track given you satiate your desire to learn and love the courses. 

System Design is an essential part of the tech-world and so will be AI. If you are good at what you do, you will be able to get a job in either fields. Irrespective of the demand/supply BS. 

Back track your life. 5 years from now how do you see yourself in the tech industry? Someone solving big scaling problems for large systems? Or someone working on complex ML models? What will give you more joy? How much of the info you already know, and what else is left for you to learn. Which companies are your dream companies? Which roles are your dream roles? 
And see what is required for your to get to these places you just thought about. And make a decision. 

I hope this helps you in analyzing and assessing. Unfortunately, I cannot decide it for you or advice you and make your decision making capability biased. 

All the best!
